src/Stats/index.jsx

```diff
@@ -1,33 +1,103 @@
+  const [currentYearSet, setCurrentYearSet] = useState(utils.getDateSet(moment(), 'years'));
+  const [view, setCurrentView] = useState('month');
+  const [overlay, setOverlay] = useState(false);
+
+
+  const ifTransition = (arr) => (arr.length !== 0);
+
+  const updateCurrent = (value, type) => {
+    switch (type) {
+      case 'month':
+        setCurrentMonthSet(utils.getDateSet(value, 'months'));
+        break;
+      case 'year':
+        setCurrentYearSet(utils.getDateSet(value, 'years'));
+        break;
+      default:
+        break;
+    }
+  };
+
+  const updateOverlay = () => {
+    if (overlay === true) {
+      setOverlay(false);
+    } else {
+      setOverlay(true);
+    }
+  };
+
+  const updateHeaderView = (type) => {
+    setCurrentView(type);
+    updateOverlay();
+  };
+
+  const updateListView = (dataList) => {
+    const current = view === 'month' ? currentMonthSet[1] : currentYearSet[1];
+    return utils.filterData(dataList, current, view);
+  };
+
   return (
-    <ScrollView>
-      <View style={styles.card}>
-        {utils.sortDataByDate(transactions).map((item, index) => (
-          <ListItem
-            key={`ListKey${index + 1}`}
-            title={item.label ? item.label : 'Undefined'}
-            subtitle={moment.unix(item.date).format('MM/DD/YYYY')}
-            rightTitle={
-              item.type === 'Income'
-                ? ` + $${item.amount}`
-                : ` - $${item.amount}`
-            }
-            leftIcon={{
-              title: 'Unknown',
-              name: 'attach-money',
-            }}
-          />
-        ))}
-      </View>
-    </ScrollView>
+    <View style={{ flex: 1, flexDirection: 'column' }}>
+      <Overlay height={200} isVisible={overlay}>
+        <View>
+          <View style={{ flexDirection: 'row', justifyContent: 'flex-end' }}>
+            <Button icon={{ name: 'close' }} type="clear" onPress={updateOverlay} />
+          </View>
+          <View style={{ marginTop: 20 }}>
+            <ListItem title="Month" topDivider bottomDivider onPress={() => updateHeaderView('month')} />
+            <ListItem title="Year" bottomDivider onPress={() => updateHeaderView('year')} />
+          </View>
+        </View>
+      </Overlay>
+      <MainHeader title="Activity" onPressBtn={updateOverlay} />
+      <SubHeader viewSet={view === 'month' ? currentMonthSet : currentYearSet} onPressBtn={updateCurrent} viewType={view} />
+      <ScrollView style={styles.deviceBody}>
+        <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
+          <LargeButton subtitle="Spent" title="-$400.00" note="Expense" bgColor="red" btnAmount={2} />
+          <LargeButton subtitle="Earned" title="+$680.00" note="Incomes" bgColor="green" btnAmount={2} />
+        </View>
+        {ifTransition(transactions)
+          ? updateListView(utils.groupData(transactions)).map((value, index) => (
+            <View style={styles.card} key={`DateGroup:${index + 1}`}>
```

Comment:
let's create some component for the card and for the transaction items, this looks too complex now

Reply:
Done